src/builtins/filters/number.rs

```diff
@@ -74,10 +74,109 @@ pub fn filesizeformat(value: &Value, _: &HashMap<String, Value>) -> Result<Value
         .map(std::result::Result::unwrap)
 }
 
+/// Formats integers using the `fmt` string given
+///
+/// Formatting options (All options passed via `format(fmt="<option>")`:
+///     * `:X` Upper hex (`42` => `2A`)
+///     * `:x` Lower hex (`42` => `2a`)
+///     * `:o` Octal (`42` => `52`)
+///     * `:b` Binary (`42` => `101010`)
+///     * `:E` Upper exponent (`42.0` => `4.2E1`)
+///     * `:e` Lower exponent (`42.0` => `4.2e1`)
+///
+/// Additionally, the `#` modifier can be passed to some formatters as well:
+///     * `:#X` Upper hex (`42` => `0x2A`)
+///     * `:#x` Lower hex (`42` => `0x2a`)
+///     * `:#o` Octal (`42` => `0o52`)
+///     * `:#b` Binary (`42` => `0b101010`)
+pub fn format(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
+    let fmt = if let Some(fmt) = args.get("fmt") {
+        try_get_value!("format", "fmt", String, fmt)
+    } else {
+        return Err(Error::msg("Filter `format` expected an arg called `fmt`"));
+    };
+    let mut chars = fmt.chars();
+
+    if !matches!(chars.next(), Some(':')) {
```

Comment:
I don't see `matches!` macro being used anywhere else in this project. Should it be an IF or MATCH ?

Reply:
`matches!()` is in the stdlib.